src/components/coin-row/TransactionCoinRow.js

```diff
@@ -73,14 +75,53 @@ const TopRow = ({ balance, pending, status, title }) => (
 );
 
 export default function TransactionCoinRow({ item, ...props }) {
-  const { contact, hash } = item;
+  const { contact } = item;
   const { network } = useAccountSettings();
   const { navigate } = useNavigation();
 
   const onPressTransaction = useCallback(async () => {
-    const { from, to, status } = item;
-    const isPurchasing = status === TransactionStatusTypes.purchasing;
-    const isSent = status === TransactionStatusTypes.sent;
+    const { hash, from, minedAt, pending, to, status, type } = item;
+
+    const calculateTimestampOfToday = () => {
+      var d = new Date();
+      d.setHours(0, 0, 0, 0);
+      return d.getTime();
+    };
+    const calculateTimestampOfYesterday = () => {
+      var d = new Date();
+      d.setDate(d.getDate() - 1);
+      d.setHours(0, 0, 0, 0);
+      return d.getTime();
+    };
+    const calculateTimestampOfThisYear = () => {
+      var d = new Date();
+      d.setFullYear(d.getFullYear(), 0, 1);
+      d.setHours(0, 0, 0, 0);
+      return d.getTime();
+    };
+    const todayTimestamp = calculateTimestampOfToday();
+    const yesterdayTimestamp = calculateTimestampOfYesterday();
+    const thisYearTimestamp = calculateTimestampOfThisYear();
+
+    const timestamp = new Date(minedAt * 1000);
+    const date = format(
```

Comment:
Actually I'd move all this into a function `getHumanReadableDate` or something like that at  `src/helpers/transactions.js`